src/cat-data-service/src/service/v1/event/objective/review_type.rs

```diff
@@ -77,51 +77,51 @@ mod tests {
-                    max: 5,
-                    map: vec![],
-                    note: None,
-                    group: Some("review_group 1".to_string()),
-                },
-                ReviewType {
-                    id: 2,
-                    name: "feasibility".to_string(),
-                    description: Some("Feasibility Rating".to_string()),
-                    min: 0,
-                    max: 5,
-                    map: vec![],
-                    note: Some(true),
-                    group: Some("review_group 2".to_string()),
-                },
-                ReviewType {
-                    id: 5,
-                    name: "vpa_ranking".to_string(),
-                    description: Some("VPA Ranking of the review".to_string()),
-                    min: 0,
-                    max: 3,
-                    map: vec![
-                        json!(
-                            {"name":"Excellent","desc":"Excellent Review"}
-                        )
-                        .to_string(),
-                        json!({"name":"Good","desc":"Could be improved."}).to_string(),
-                        json!({"name":"FilteredOut","desc":"Exclude this review"}).to_string(),
-                        json!({"name":"NA", "desc":"Not Applicable"}).to_string()
-                    ],
-                    note: Some(false),
-                    group: None,
-                }
-            ])
+            serde_json::Value::from_str(
+                String::from_utf8(response.into_body().data().await.unwrap().unwrap().to_vec())
+                    .unwrap()
+                    .as_str()
+            )
             .unwrap(),
-            String::from_utf8(response.into_body().data().await.unwrap().unwrap().to_vec())
-                .unwrap()
+            serde_json::json!(
+                [
+                    {
+                        "id": 1,
+                        "name": "impact",
+                        "description": "Impact Rating",
+                        "min": 0,
+                        "max": 5,
+                        "map": [],
+                        "group": "review_group 1",
+                        "note": null,
+                    },
+                    {
+                        "id": 2,
+                        "name": "feasibility",
+                        "description": "Feasibility Rating",
+                        "min": 0,
+                        "max": 5,
+                        "map": [],
+                        "note": true,
+                        "group": "review_group 2"
+                    },
+                    {
+                        "id": 5,
+                        "name": "vpa_ranking",
+                        "description": "VPA Ranking of the review",
+                        "min": 0,
+                        "max": 3,
+                        "map": [
+                            {"name": "Excellent", "desc": "Excellent Review"},
```

Comment:
If this is what your proposing, then the API should explicitly state its "name" and "desc" what they mean and if they are optional. ("desc" should be optional, "name" not).

Reply:
Yes, will add a specific types and descriptions into the API doc, but will not modify db schema and how I am processing its, will let it as general as possible (array of json objects).